### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by hoythan**
thank you for your advice.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    /**
+     * Returns a Promise for the value returned from input.
+     *
+     * @param {string | function} input - .
+     * If a function, it's stringified and passed to JXA. It should be atomic, meaning it doesn't access anything outside its body.
+     * If a string, you can access the specified arguments with args array. Use the arguments parameter rather than template interpolation so you don't have to do escaping.
+     * @param {!Array[any]} args - Arguments to pass to the JXA context. Items should be serializable (JSON.stringify'able).
+     * @returns Returns a Promise for the value returned from input.
+     */
+    (input: string | ((...args:Array<string>) => void), args?: Array<any>):Promise<any>;
```
Comment (SamVerschueren) [inline index.d.ts]: I would use [function overloads](https://www.typescriptlang.org/docs/handbook/functions.html#overloads). Then you can define 2 different signatures for the same method.

```ts
type JSONStringifyable = object | number | string | boolean;

(input: string, args?: JSONStringifyable[]): Promise<JSONStringifyable>;
<T, U>(input: (...args: U[]) => T, args: U[]): Promise<T>;
```

Something like that, needs some more finetuning perhaps but it's a good start.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export interface runJxa {
+    /**
+     * Returns a Promise for the value returned from input.
+     *
+     * @param {string | function} input - .
+     * If a function, it's stringified and passed to JXA. It should be atomic, meaning it doesn't access anything outside its body.
+     * If a string, you can access the specified arguments with args array. Use the arguments parameter rather than template interpolation so you don't have to do escaping.
+     * @param {!Array[any]} args - Arguments to pass to the JXA context. Items should be serializable (JSON.stringify'able).
```
Comment (sindresorhus) [inline index.d.ts]: > @param should not include the parameter type. - https://github.com/sindresorhus/typescript-definition-style-guide

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+export interface runJxa {
+    /**
+     * Returns a Promise for the value returned from input.
+     *
+     * @param {string | function} input - .
```
Comment (sindresorhus) [inline index.d.ts]: Is `- .` a convention? I haven't seen that before. Are you sure the below lines will show up as description for `input`?

File: index.d.ts (hunk 1 above)
Comment (sindresorhus) [reply index.d.ts]: Use shorthand syntax for array.

**Comment by hoythan** [comment]
please, I really need typescript support。
If there is anything wrong, could you please help me revise it?
🙏🙏

**Comment by sindresorhus** [comment]
@hoythan Sorry, there's a lot to comment about and I have little time right now. Read through https://github.com/sindresorhus/typescript-definition-style-guide and make sure you follow it.

File: index.d.ts [hunk 4]
```diff
@@ -1,27 +1,26 @@
-type JSONStringifyable = object | number | string | boolean;
```
Comment (sindresorhus) [inline index.d.ts]: Why are you removing this?

**Comment by hoythan** [comment]
you are helping me confirm

**Comment by sindresorhus** [comment]
There are still many things not followed from https://github.com/sindresorhus/typescript-definition-style-guide And you have not kept the comments in sync with the actual code.

**Comment by sindresorhus** [comment]
Closing for lack of activity and response.

**Pull request closed without merging**